src/Microsoft.DotNet.Interactive.HttpRequest.Tests/ParserTests.cs

```diff
@@ -0,0 +1,645 @@
+
+using System.Linq;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Microsoft.DotNet.Interactive.HttpRequest.Tests.Utility;
+using Xunit;
+
+namespace Microsoft.DotNet.Interactive.HttpRequest.Tests;
+
+public class ParserTests
+{
+
+    public class Trivia
+    {
+        [Fact]
+        public void it_can_parse_an_empty_string()
+        {
+            var result = HttpRequestParser.Parse("");
+            result.SyntaxTree.Should().BeNull();
+
+            // TODO: Test error reporting.
+        }
+
+        [Fact]
+        public void it_can_parse_a_string_with_only_whitespace()
+        {
+            var result = HttpRequestParser.Parse(" \t ");
+
+            result.SyntaxTree.RootNode
+                  .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which
+                  .MethodNode.ChildTokens.First().TextWithTrivia.Should().Be(" \t ");
+
+            // TODO: Test error reporting.
+        }
+
+        [Fact]
+        public void it_can_parse_a_string_with_only_newlines()
+        {
+            var result = HttpRequestParser.Parse("\r\n\n\r\n");
+
+            result.SyntaxTree.RootNode
+                  .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which
+                  .MethodNode.TextWithTrivia.Should().Be("\r\n\n\r\n");
+
+            // TODO: Test error reporting.
+            // Regardless of number of newlines, there should be consistent nodes
+            // set of whitespace, newline, and punctuation
+            //each token kind
+            //combination of characters should be checked by the lexer
+        }
+    }
+    
+    
+    public class Lexer
+    {
+        [Fact]
+
+        public void multiple_whitespaces_are_treated_as_a_single_token()
+        {
+            var result = HttpRequestParser.Parse("     ");
+
+            using var _ = new AssertionScope();
+
+            result.SyntaxTree.RootNode
+            .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which
+            .MethodNode.ChildTokens.First().Should().BeOfType<HttpSyntaxToken>();
+
+            result.SyntaxTree.RootNode
+            .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which
+            .MethodNode.ChildTokens.First().TextWithTrivia.Should().Be("     ");
+        }
+    }
+    
+    public class Method
+    {
+        [Fact]
+        public void whitespace_is_legal_at_the_beginning_of_a_request()
+        {
+            var result = HttpRequestParser.Parse("  GET https://example.com");
+
+            result.SyntaxTree.RootNode
+                  .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which
+                  .MethodNode.ChildTokens.First().TextWithTrivia.Should().Be("  ");
+        }
+
+        [Fact]
+        public void newline_is_legal_at_the_beginning_of_a_request()
+        {
+            var result = HttpRequestParser.Parse(
+                """
+        
+        GET https://example.com
+        """);
+
+            result.SyntaxTree.RootNode
```

Comment:
I would assert that the leading trivia contains newline tokens rather than on the specific text of those tokens, which might vary (and is causing this test to fail on Linux).